steve/runner.go

```diff
@@ -3,64 +3,81 @@ package steve
 import (
 	"bytes"
 	"context"
-	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/google/uuid"
 	"github.com/mailgun/holster/v4/collections"
 	"github.com/mailgun/holster/v4/syncutil"
+	"github.com/pkg/errors"
 )
 
 var (
 	ErrJobNotFound   = errors.New("no such job found")
 	ErrJobNotRunning = errors.New("job not running")
+	ErrTooManyJobs   = errors.New("too many jobs")
 )
 
 type jobIO struct {
-	br      syncutil.Broadcaster
-	writer  io.WriteCloser
-	mutex   sync.Mutex
-	buffer  bytes.Buffer
-	id      ID
-	running int64
-	job     Job
+	sync.RWMutex
+	br       syncutil.Broadcaster
+	writer   io.WriteCloser
+	buffer   bytes.Buffer
+	id       ID
+	job      Job
+	status   Status
+	stopChan chan struct{}
```

Comment:
Is it to stop the job or to notify that it was stopped?

Reply:
Notify that it was stopped, any routines what want to know if the job was stopped will listen to this channel for a close.

Reply:
Yeah. I'm just confused since this naming is used for stopping something in other services, not about notifying. 
What about naming it something like `stoppedChan`?